src/App.tsx

```diff
@@ -1,11 +1,18 @@
-import { useState } from 'react';
-import './App.scss';
+import { useCallback, useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
+import './App.scss';
+
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = useCallback((movie: Movie) => {
+    if (!movies.some(film => film.imdbId === movie.imdbId)) {
+      setMovies([...movies, movie]);
```

Comment:
use functional setState here

Reply:
please fix it `setMovies(prevState => [...prevState, movie]);`